hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionImplementation.java

```diff
@@ -1008,9 +1008,11 @@ private RegionLocations locateRegionInMeta(TableName tableName, byte[] row, bool
       }
       // Query the meta region
       long pauseBase = connectionConfig.getPauseMillis();
-      takeUserRegionLock();
       final long lockStartTime = EnvironmentEdgeManager.currentTime();
```

Comment:
this should be right after the `takeUserRegionLock()`call.

Reply:
OK.

Reply:
So we are modifying behaviour for the `lockStartTime`? We should probably rename it, then, because what we are recording here now is not the lock start time anymore, but the time before locking.

Reply:
Thanks, you are right. the lockStartTime here marks the timestamp after grabbing the lock(after the takeUserRegionLock() call), while the takeUserRegionLock() has already recorded the lock waiting time.